core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java

```diff
@@ -176,6 +177,9 @@ public boolean send(Transmission transmission) {
                 if (ioe instanceof ConnectTimeoutException) {
                     transmissionPolicyManager.suspendInSeconds(TransmissionPolicy.BLOCKED_BUT_CAN_BE_PERSISTED, DEFAULT_BACKOFF_TIME_SECONDS);
                 }
+            } catch (IllegalStateException e) {
+                transmissionPolicyManager.stop(SHUTDOWN_TIME, TimeUnit.SECONDS);
```

Comment:
@andresol , Is there a way this code (or even app) can get out of the illegal state with one of the next method invocations being successful?

I know I suggested `stop()` myself, but, on the second thought, I now realize that it leads to all sender threads being stopped and never trying again if they are not re-initialized. So, a single `IllegalStateException` will stop telemetry from the app up until the next app restart. With this, if there is a possibility that the code may recover on its own - we'd better let it try (unobtrusively of course, without infinite loop :) )

Sorry that it turned out to be a lengthy discussion with back and forth, I just do not want to jump into the conclusion with the severe side effects. Let me know your thoughts.